ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsPruning.scala

```diff
@@ -90,246 +95,349 @@ private[backend] trait StorageBackendTestsPruning
-        consuming = Vector(EventConsuming(2)),
-        consumingFilterStakeholder = Vector(
-          FilterConsumingStakeholder(2, 2),
-          FilterConsumingStakeholder(2, 3),
-        ),
-        txMeta = Vector(TxMeta("00000001"), TxMeta("00000002")),
-      )
+    def assertAllDataPresent(): Assertion = assertIndexDbDataSql(
+      create = Vector(EventCreate(1)),
+      createFilterStakeholder = Vector(
+        FilterCreateStakeholder(1, 2),
+        FilterCreateStakeholder(1, 3),
+      ),
+      createFilterNonStakeholder = Vector(FilterCreateNonStakeholder(1, 4)),
+      consuming = Vector(EventConsuming(2)),
+      consumingFilterStakeholder = Vector(
+        FilterConsumingStakeholder(2, 2),
+        FilterConsumingStakeholder(2, 3),
+      ),
+      txMeta = Vector(TxMeta("00000010"), TxMeta("00000011")),
     )
 
-    // Prune
+    executeSql(updateLedgerEnd(offset(2), 2L))
+    assertAllDataPresent()
+    // Prune at the offset of the create event
+    pruneEventsSql(offset(10), pruneAllDivulgedContracts = true)
+    assertAllDataPresent()
+    // Prune at the offset of the archive event
+    pruneEventsSql(offset(11), pruneAllDivulgedContracts = true)
+    assertIndexDbDataSql()
+  }
+
+  it should "prune participant_transaction_meta table when all its contracts are archived" in {
+    // A transaction with two contracts
+    val create1 = dtoCreate(
+      offset = offset(10),
+      eventSequentialId = 1L,
+      contractId = hashCid("#1"),
+      signatory = signatoryParty,
+      observer = observerParty,
+      nonStakeholderInformees = Set(nonStakeholderInformeeParty),
+    )
+    val create2 = dtoCreate(
+      offset = offset(10),
+      eventSequentialId = 2L,
+      contractId = hashCid("#2"),
+      signatory = signatoryParty,
+      observer = observerParty,
+      nonStakeholderInformees = Set(nonStakeholderInformeeParty),
+    )
+    // A transaction with an archive of the first contract
+    val archive1 = dtoExercise(
+      offset = offset(11),
+      eventSequentialId = 3L,
+      consuming = true,
+      contractId = hashCid("#1"),
+      signatory = signatoryParty,
+    )
+    // A transaction with an archive of the second contract
+    val archive2 = dtoExercise(
+      offset = offset(12),
+      eventSequentialId = 4L,
+      consuming = true,
+      contractId = hashCid("#2"),
+      signatory = signatoryParty,
+    )
+    executeSql(backend.parameter.initializeParameters(someIdentityParams))
+    // Ingest a create and archive event
     executeSql(
-      backend.event.pruneEvents(offset(2), pruneAllDivulgedContracts = true)(
+      ingest(
+        Vector(
+          // Allocating parties so that the contracts we create later are not considered to be a case of immediate divulgence
+          dtoPartyEntry(offset(1), signatoryParty),
+          dtoPartyEntry(offset(2), observerParty),
+          dtoPartyEntry(offset(3), nonStakeholderInformeeParty),
```

Comment:
are not these parties already created here: https://github.com/digital-asset/daml/pull/16257/files#diff-27299a8458dbbbcf2ae3bd2cda78e4afc4e9e5cb7cbfa5ea646bc07e5cf32ab8R201-R203?

Reply:
These are the same but created in two separate test cases. After each test case we reset the DB contents